File: SurperPm-agent-web/backend/app/models/goal.py
"""Goal table model."""

from datetime import UTC, datetime

from sqlmodel import Field, SQLModel


class Goal(SQLModel, table=True):
    __tablename__ = "goal"

    id: int | None = Field(default=None, primary_key=True)
    workspace_id: str = Field(foreign_key="workspace.id", nullable=False)
    title: str = Field(nullable=False)
    description: str | None = Field(default=None)
    status: str = Field(default="todo")  # todo | doing | review | done | failed
    priority: int = Field(default=0)
    assigned_to: str | None = Field(default=None)
    suggested_assignee: str | None = Field(default=None)
    parent_goal_id: int | None = Field(default=None, foreign_key="goal.id")
    group_id: int | None = Field(default=None, foreign_key="goal_group.id")
    token_budget: int | None = Field(default=None)
    session_name: str | None = Field(default=None)
    slug: str | None = Field(default=None)
    repo_url: str | None = Field(default=None)
    repo_path: str | None = Field(default=None)
    repos: str | None = Field(default=None)
    reviewed_by: str | None = Field(default=None)
    deadline: str | None = Field(default=None)
    reviewed_at: datetime | None = Field(default=None)
    created_at: datetime = Field(default_factory=lambda: datetime.now(UTC))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(UTC))

File: SurperPm-agent-web/frontend/src/lib/schemas/goal.ts
import { z } from "zod";

export const goalSchema = z.object({
  id: z.number(),
  workspace_id: z.string(),
  title: z.string(),
  description: z.string().nullable(),
  status: z.enum(["todo", "doing", "review", "done", "failed"]),
  reviewed_by: z.string().nullable().optional(),
  reviewed_at: z.string().nullable().optional(),
  priority: z.number(),
  assigned_to: z.string().nullable().optional(),
  suggested_assignee: z.string().nullable().optional(),
  parent_goal_id: z.number().nullable().optional(),
  token_budget: z.number().nullable().optional(),
  slug: z.string().nullable().optional(),
  repo_url: z.string().nullable().optional(),
  repo_path: z.string().nullable().optional(),
  repos: z.string().nullable().optional(),
  created_at: z.string(),
  updated_at: z.string(),
});

export type Goal = z.infer<typeof goalSchema>;

export const goalListSchema = z.array(goalSchema);
